src/main/java/seedu/address/logic/commands/ExportCommand.java

```diff
@@ -0,0 +1,223 @@
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data \n"
+            + "Example 2: " + COMMAND_WORD + " " + PREFIX_NAME + "Exportfile " + PREFIX_RANGE + "1 "
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data \n"
+            + "Example 3: " + COMMAND_WORD + " " + PREFIX_NAME + "Exportfile " + PREFIX_RANGE + "1,2 "
+            + PREFIX_TAG_EXPORT + " friends " + PREFIX_PATH + "/src/main/data";
+
+
+    private Tag tag;
+    private final String range;
+    private final String path;
+    private AddressBook teachConnectBook;
+    private AddressBookStorage teachConnectStorage;
+    private final String nameOfExportFile;
+
+    /**
+     * Creates an ExportCommand to export the specified {@code Persons}
+     */
+    public ExportCommand(String range, Tag tag, String path, String nameOfExportFile) {
+        requireNonNull(range);
+        requireNonNull(tag);
+        requireNonNull(path);
+        requireNonNull(nameOfExportFile);
+
+        this.range = range;
+        this.path = path;
+        this.tag = tag;
+        this.nameOfExportFile = nameOfExportFile;
+
+        teachConnectBook = new AddressBook();
+    }
+
+    @Override
+    public CommandResult execute() {
+        String[] rangeGiven;
+        try {
+            rangeGiven = handleRange();
+        } catch (IOException e) {
+            return new CommandResult(MESSAGE_RANGE_ERROR);
+        }
+        if (rangeGiven[0].equals("all")) {
+            try {
+                exportAllRange(tag);
+            } catch (DuplicatePersonException e) {
+                throw new AssertionError(MESSAGE_FAIL);
+            }
+        } else {
+            if (rangeGiven.length != 1) {
+                for (int i = 0; i < rangeGiven.length; i++) {
+                    int low = Integer.parseInt(rangeGiven[0]);
+                    int high = Integer.parseInt(rangeGiven[1]);
+                    if (low >= high) {
+                        return new CommandResult(MESSAGE_RANGE_ERROR);
+                    } else {
+                        try {
+                            exportRange(low, high, tag);
+                        } catch (DuplicatePersonException e) {
+                            throw new AssertionError(MESSAGE_FAIL);
+                        } catch (IndexOutOfBoundsException e) {
+                            return new CommandResult(MESSAGE_OUT_OF_BOUNDS);
+                        }
+                    }
+                }
+            } else {
+                int low = Integer.parseInt(rangeGiven[0]);
+                try {
+                    exportSpecific(low);
+                } catch (DuplicatePersonException e) {
+                    throw new AssertionError(MESSAGE_FAIL);
+                } catch (IndexOutOfBoundsException e) {
+                    return new CommandResult(MESSAGE_OUT_OF_BOUNDS);
+                }
+            }
+
```

Comment:
You may want to refactor this block of code. Apply SLAP and consider using else if instead of nested conditional statement.